src/Spectre.AngularClient/src/app/upload/upload.component.spec.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * upload.component.spec.ts
+ * Unit tests for single upload component.
+ *
+ Copyright 2018 Sebastian Pustelnik
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Http, BaseRequestOptions, Response, ResponseOptions } from '@angular/http';
+import { MockBackend} from '@angular/http/testing';
+
+import { RouterTestingModule } from '@angular/router/testing';
+import { UploadComponent } from "./upload.component";
+import { UploadService } from "./shared/upload.service";
+import {FormsModule} from "@angular/forms";
+
+
+describe('UploadComponent', () => {
```

Comment:
What about tests against enabling & disabling the button in the case of valid / invalid data?